### PR TITLE
Einsteinathome upstream

### PR DESCRIPTION
**Pull request by bema-ligo**
Push the changes for running PyCBC_inspiral on Einstein@Home upstream.

This adds a directory tools/einsteinathome to the source, containing a bit of code useful for Einstein@Home.

The behavior of pycbc_inspiral under normal circumstances is unchanged as long as the environment variables PROJECT_DIR, NO_PKGCONFIG, NO_GETCONF, LEVEL2_CACHE_SIZE, INITIAL_LOG_LEVEL and WEAVE_FLAGS are not set and the command-line option --fixed-weave-cache is not given. If set, the behavior is modified in the way described in the individual commit messages.

### CODE REVIEW

**Comment by ahnitz** [comment]
Thank you! I'll take a more detailed look at this tomorrow.

**Comment by josh-willis** [comment]
Hi,

In `__init__.py`, s there a reason that after you've set `WEAVE_FLAGS` from the environment, and possibly changed `DEFAULT_WEAVE_FLAGS` based on what's in `WEAVE_FLAGS` (so after line 151), that you don't just do 

`WEAVE_FLAGS += DEFAULT_WEAVE_FLAGS`

I think then you wouldn't have had to change so many individual files to update their compilation flags, and then also if someone writes some new code, they won't have to remember to write `WEAVE_FLAGS + DEFAULT_WEAVE_FLAGS`.

**Comment by bema-ligo** [comment]
@josh-willis : What I needed was a way to switch off the "`-march=native`" that is hardcoded in many files, in order to avoid conflicts with a possibly different architecture specified in the environment `WEAVE_FLAGS`. So changing that many files seems unavoidable.

However, indeed it looks easier to delete the constant phrases "` + '-march=native -O3 -w`'" in that fies completely and just manipulate `WEAVE_FLAGS` in `__init__.py` once, which would also save the hazzle of extra imports.

**Comment by ahnitz** [comment]
@bema-ligo You can just update this pull request with a new patch actually. (or force push to this branch if you prefer).

**Comment by bema-ligo** [comment]
@ahnitz : branch updated.

File: pycbc/waveform/bank.py [hunk 1]
```diff
@@ -68,6 +68,11 @@ class BaseFilterBank(object):
     about elements of an xml template bank.
     """
     def __init__(self, filename, approximant=None, **kwds):
+
```
Comment (ahnitz) [inline pycbc/waveform/bank.py]: I don't understand why the following is required. Why don't the command line options to pycbc_inspiral contain the correct path (either relative or absolute) already?

Comment (bema-ligo) [inline pycbc/waveform/bank.py]: Indeed this modification looks pretty convoluted and backwards. However I found no other way to make this work with BOINC's handling of files communicated with the server ("input" and "output" files), the limited abilities of the BOINC "wrapper" program to manipulate the command-line of the program it calls, and the fact that a Cygwin Pyinstaller bundle can't (relatively) reference files outside the "root" directory of the bundle.

Comment (bema-ligo) [inline pycbc/waveform/bank.py]: Knowing a bit more now I'm currently making another attempt at doing the input file handling without any modifications to PyCBC, and without that patch in particular.

**Comment by bema-ligo** [comment]
I found a way of setting up BOINC "jobs" and the wrapper that doesn't require patching the file handling of PyCBC. I updated the branch, removing the patch in question.

**Comment by ahnitz** [comment]
@lppekows I'm no longer an expert on the pyinstaller bundling. Can you verify that the changes here to hook-pycbc and hooky-weave will not have any effects for the standard bundles. I wouldn't think so.

**Comment by lppekows** [comment]
I applied the patches to the hook files on top of 1.3.7, everything built and ran successfully.

**Comment by ahnitz** [comment]
Great, thanks Larne.

**Pull request merged**